slick/src/main/scala/slick/jdbc/SQLServerProfile.scala

```diff
@@ -248,29 +255,143 @@ trait SQLServerProfile extends JdbcProfile {
+          case null => null
+          case serializedTime =>
+
+            val sep = serializedTime.indexOf('.')
+            if (sep == -1) {
+              Time.valueOf(serializedTime).toLocalTime
+            } else {
+              LocalTime.parse(serializedTime, formatter)
+            }
+        }
+      }
+      override def valueToSQLLiteral(value: LocalTime) = {
+        value match {
+          case null => "NULL"
+          case _ => s"(convert(time(6), '$value'))"
         }
       }
     }
     class TimestampJdbcType extends super.TimestampJdbcType {
-      /* TIMESTAMP in SQL Server is a data type for sequence numbers. What we
-       * want here is DATETIME. */
-      override def sqlTypeName(sym: Option[FieldSymbol]) = "DATETIME"
+      override def sqlTypeName(sym: Option[FieldSymbol]) = "DATETIME2(6)"
       override def valueToSQLLiteral(value: Timestamp) = "(convert(datetime, {ts '" + value + "'}))"
     }
+    class LocalDateTimeJdbcType extends super.LocalDateTimeJdbcType {
+      private[this] val formatter : DateTimeFormatter = {
+        new DateTimeFormatterBuilder()
+          .append(DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss"))
+          .optionalStart()
+          .appendFraction(ChronoField.NANO_OF_SECOND, 0, 6, true)
+          .optionalEnd()
+          .toFormatter()
+      }
+      override def sqlTypeName(sym: Option[FieldSymbol]) = "DATETIME2(6)"
+      override def getValue(r: ResultSet, idx: Int): LocalDateTime = {
+        r.getTimestamp(idx) match {
+          case null =>
+            null
+          case timestamp =>
+            timestamp.toLocalDateTime
+        }
+      }
+    }
+    class InstantJdbcType extends super.InstantJdbcType {
+      private[this] val formatter : DateTimeFormatter = {
+        new DateTimeFormatterBuilder()
+          .append(DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss"))
+          .appendFraction(ChronoField.NANO_OF_SECOND, 0, 6, true)
+          .appendPattern(" ")
+          .appendOffset("+HH:MM", "")
+          .toFormatter()
+      }
+      private[this] def serializeInstantValue(value : Instant) : String = {
+        formatter.format(
+          OffsetDateTime.ofInstant(value, ZoneOffset.UTC)
+        )
+      }
+      override def sqlTypeName(sym: Option[FieldSymbol]) = "DATETIMEOFFSET(6)"
+      override def setValue(v: Instant, p: PreparedStatement, idx: Int) : Unit = {
+        p.setString(idx, serializeInstantValue(v))
+      }
+      override def updateValue(v: Instant, r: ResultSet, idx: Int) : Unit = {
+        r.updateString(idx, serializeInstantValue(v))
+      }
+
+      override def getValue(r: ResultSet, idx: Int): Instant = {
+        r.getString(idx) match {
+          case null =>
+            null
+          case dateStr =>
+            OffsetDateTime.parse(dateStr, formatter).toInstant()
+        }
+      }
+      override def valueToSQLLiteral(value: Instant) = {
+        value match {
+          case null =>
```

Comment:
this is the only profile that handles `value == null`. Why don't the other do it?

Reply:
The `case null =>` couldn't be reached. They've been removed.